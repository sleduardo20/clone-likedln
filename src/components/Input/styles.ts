import styled, { css } from 'styled-components';
import { InputProps } from '.';

type ContainerProps = Pick<InputProps, 'icon' | 'name'>;

export const Container = styled.div<ContainerProps>`
  ${({ theme, icon }) => css`
    width: 100%;
    padding: ${theme.spacings.xsmal} ${theme.spacings.medium};
    height: ${theme.spacings.huge};
    background: ${theme.colors.white};

    display: flex;
    align-items: center;
    justify-content: center;

    ${!!icon &&
    css`
      > svg {
        width: ${theme.spacings.large};
        height: ${theme.spacings.large};
        margin-right: ${theme.spacings.small};
        color: ${theme.colors.blue};
      }
    `}

    > input {
      flex: 1;
      outline: none;
      padding: ${theme.spacings.small};
      font-size: ${theme.font.size.normal};
      line-height: ${theme.font.size.medium};
      color: ${theme.colors.black};
      border: none;
      width: 100%;
      height: 100%;

      &::placeholder {
        color: ${theme.colors.gray};
      }
    }
  `}
`;
