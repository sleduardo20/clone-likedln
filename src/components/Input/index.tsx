import { InputHTMLAttributes } from 'react';
import { Container } from './styles';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  name: string;
  icon?: React.ReactNode;
}

const Input = ({ name, icon, ...rest }: InputProps) => {
  return (
    <Container name={name} icon={icon}>
      {!!icon && icon}
      <input type="text" {...rest} />
    </Container>
  );
};

export default Input;
