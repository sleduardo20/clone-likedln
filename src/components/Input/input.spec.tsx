import { screen } from '@testing-library/react';
import { FiSearch } from 'react-icons/fi';
import { renderWithTheme } from '../../utils/renderWithTheme';

import Input from '.';

describe('<Input/>', () => {
  it('should be able render Input compoment correctly', () => {
    const { container } = renderWithTheme(<Input name="search" />);

    expect(screen.getByRole('textbox').parentElement).toHaveAttribute(
      'name',
      'search',
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able render Input compoment correctly with icon', () => {
    renderWithTheme(
      <Input name="search" icon={<FiSearch data-testid="icon" />} />,
    );

    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
