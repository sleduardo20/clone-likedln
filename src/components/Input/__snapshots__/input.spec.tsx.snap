// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should be able render Input compoment correctly 1`] = `
.c0 {
  width: 100%;
  padding: 0.4rem 1.6rem;
  height: 5.4rem;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > input {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  outline: none;
  padding: 0.8rem;
  font-size: 1.6rem;
  color: #181818;
  border: none;
  width: 100%;
  height: 100%;
}

.c0 > input::-webkit-input-placeholder {
  color: #CECECE;
}

.c0 > input::-moz-placeholder {
  color: #CECECE;
}

.c0 > input:-ms-input-placeholder {
  color: #CECECE;
}

.c0 > input::placeholder {
  color: #CECECE;
}

<div
  class="c0"
  name="search"
>
  <input
    type="text"
  />
</div>
`;
